src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,88 @@
+package tasks;
+
+/**
+ * Abstract class representing a generic task in Bento.
+ * Each task has a name and a status indicating whether it is done or not.
+ * Subclasses are expected to provide task-specific implementations.
+ */
+public abstract class Task {
+    private String taskName;
+    private boolean isDone = false;
+
+    /**
+     * Constructs a Task with the specified task name.
+     *
+     * @param taskName The name of the task.
+     */
+    public Task(String taskName) {
+        this.taskName = taskName;
+    }
+
+    /**
+     * Returns the name of the task.
+     *
+     * @return The task's name.
+     */
+    public String getTaskName() {
+        return taskName;
+    }
+
+    /**
+     * Sets the task's name.
+     *
+     * @param taskName The new name for the task.
+     */
+    public void setTaskName(String taskName) {
+        this.taskName = taskName;
+    }
+
+    /**
+     * Returns the completion status of the task.
+     *
+     * @return True if the task is done, false otherwise.
+     */
+    public boolean isDone() {
+        return isDone;
+    }
+
+    /**
+     * Marks the task as done or undone.
+     *
+     * @param done True to mark the task as done, false to mark it as undone.
+     */
+    public void setDone(boolean done) {
+        isDone = done;
+    }
+
+    /**
+     * Returns a string marker indicating whether the task is done or not.
+     * A done task is marked with "[x]", and an undone task is marked with "[ ]".
+     *
+     * @return The done marker as a string.
+     */
+    public String getDoneMarker() {
```

Comment:
The getDoneMarker() method can be simplified using a ternary operator to improve readability and reduce verbosity.